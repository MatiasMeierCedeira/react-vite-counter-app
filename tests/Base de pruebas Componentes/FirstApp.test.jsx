import { render } from "@testing-library/react"
import { FirstApp } from "../../src/FirstApp";


    describe('Pruebas First App',()=>
        {
            test('Validación de parametros con snapshot',()=>
                {
                  const title ='Soy el Maty'; 
                  const {container} = render(<FirstApp title = {title} />)
                  expect(container).toMatchSnapshot()

                })

                // Pruebas de snapShot . No son muy recomendables cuando estamos en una etapa de desarrollo. En ambientes mas altos si.
                // Las pruebas siempre tienen que ser lo mas flexible posibles. Se debe evaluar la consistencia de la información para que
                // sea dinamicas las pruebas.
                test('Debemos mostrar el titulo en H1',()=>
                  {
                    const title ='Soy el Maty'; 
                    const {container,getByText,getByTestId} = render(<FirstApp title = {title} />);
                    expect(getByText(title)).toBeTruthy();

                    const h1 = container.querySelector('h1');
                    expect(h1.innerHTML).toContain(title);

                    // prueba con GetTextById
                    expect(getByTestId('test-title')).toContain(title);


                  });


        })