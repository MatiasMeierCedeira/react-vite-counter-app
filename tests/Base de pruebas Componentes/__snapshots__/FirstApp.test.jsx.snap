// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas First App Validaci√≥n de parametros con snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Soy el Maty
  </h1>
  <p>
    Fernando Herrera
  </p>
</div>
`;
